src/test/parse.ts

```diff
@@ -114,6 +118,25 @@ export function parseErrorsFromMarkup(text: string): LintError[] {
     return lintErrors;
 }
 
+function substituteMessage(substitutions: Map<string, string>, message: string): string {
+    const substitution = substitutions.get(message);
+    if (substitution !== undefined) {
+        return substitution;
+    }
+    return formatMessage(substitutions, message);
+}
+
+function formatMessage(substitutions: Map<string, string>, message: string): string {
+    const formatMatch = /^([\w_]+) % \(\s*([^,]+(?:\s*,\s*[^,]+)*)\s*\)$/.exec(message);
+    if (formatMatch !== null) {
+        const base = substitutions.get(formatMatch[1]);
+        if (base !== undefined) {
+            message = format(base, ...(formatMatch[2].trim().split(/\s*,\s*/g)));
```

Comment:
Format arguments now need quotes. The implementation uses typescript's scanner to parse them. Escaping also works.